### ISSUE TITLE
The label Extraction

### ISSUE THREAD

**Issue by HaishuoFang** [zcakhaa/DeepLOB-Deep-Convolutional-Neural-Networks-for-Limit-Order-Books#11]
Hello,

I have a question about the labelling? When you label data, you use the next k timesteps avg price / current price. Are k timesteps here natural timesteps or the next k events(some changes happening in LOB)?

Thank you

**Comment by HaishuoFang** [comment]
Another question is about input, What is the meaning of the most 100 updates? Does it mean the most 100 events

**Comment by zcakhaa** [comment]
It is the tick time (event) not clock time. Yes, the most 100 events.


________________________________

**Comment by HaishuoFang** [comment]
Thank you for your reply! I am adapting your model to my dataset. I have 800,000 events per day and one-month data, which is a huge dataset. It is really time-consuming to run one epoch. The weird thing is f1 of class "down" is zero, which means it cannot learn anything information about class "down" and loss don't coverage. I'll inspect it. 

I just wonder if you have similar experience before? Thank you

**Comment by zcakhaa** [comment]
I haven't experienced it. If the loss does not improve on training data, there must be anything wrong with the setup.

________________________________

**Comment by HaishuoFang** [comment]
Thanks for your reply. I solved this problem by adding the BatchNorm layer. It makes the model more robust and easy to train

**Comment by zcakhaa** [comment]
Great. You can also try different activation functions. Sometimes, ReLU or Leaky-ReLU would lead to unstable training for time-series.


________________________________

**Comment by HaishuoFang** [comment]
The final question: what is the frequency of high-frequency trading? I got 0.017 seconds lob data, which is quite large. I downsampled it on half-minute. I wonder if the frequency of half-minute is too low? Do you know what is the suitable time interval in high-frequency trading? Thank you

**Comment by zcakhaa** [comment]
It depends on your setup. For doing market-making, you probably need to work with the raw tick time without downsampling. If you are paying for the spread, you can downsample it to half-minute or minute.

________________________________

**Comment by HaishuoFang** [comment]
Thanks for your reply. Does it make sense to compare the performance of different sample frequency? It seems the only difference between them is long-term or short-term price movements

**Comment by zcakhaa** [comment]
As I mentioned that it depends on your setup. If you do market-making, you are earning spread instead of paying for it. If you need to pay for the spread, you probably need a lower frequency so the price movement is big enough to cover the costs.

________________________________

**Comment by HaishuoFang** [comment]
Thanks for your patience. Got it!
